### ISSUE TITLE
add vertical-rl Panel View/Virtual Panel in Landscape mode (2 page layout manga)

### ISSUE THREAD

**Issue by mmmeu** [ciromattia/kcc#667]
Can you help me with reading manga on landscape mode with Panel View in two page layout?

I modified the comic2ebook.py file, but unfortunately the Panel View goes like:
full two page -> first page top -> second page top -> first page middle -> second page middle, first page bottom -> second page bottom -> next page.

I tried:
1. Setting the vertical-rl to the metadata primary-writing-mode, modifying the boxStyles like the picture below.
2. Removing the app-amzn-magnify, region-mag, and all PV divs, but with no success (virtual panel doesn't show up).
3. Removing all Spread lines, so It is with the correct top, middle, bottom Panel View, but it shows only one page.
4. Combining the two page in one image file, disabled the processing, so it works like in the picture but the page turn is slow and I don't know how to automate the image combining process.

So is there a way to do this vertical Panel View while keeping the spread, two page layout?

![Screenshot 2024-01-13 at 13-13-44 AmazonKindlePublishingGuidelines pdf](https://github.com/ciromattia/kcc/assets/126781530/09dcf3f5-a18c-470e-8e8d-0fb3017f40ff)

**Comment by VampiroMedicado** [comment]
What happends if you force the following condition? https://github.com/ciromattia/kcc/blob/33fed662febc610ed886398acf3b73a1f2f1a050/kindlecomicconverter/comic2ebook.py#L170-L178

In theory, based on the order selected you should see the panel moving only right to left or viceversa.

---

If you want to have the position 2 and 5, then you should add PV-ML and PV-MR with the correct order for each case.

If the Kindle you use has a rotation system (my Paperwhite 11 doesn't have that) did you try to use add a media query? https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation

**Comment by axu2** [comment]
Going to add the relevant KDP article here: https://kdp.amazon.com/en_US/help/topic/G9GSTY4LTRT39D4Z#comics

**Comment by mmmeu** [comment]
> What happends if you force the following condition?
> 
> https://github.com/ciromattia/kcc/blob/33fed662febc610ed886398acf3b73a1f2f1a050/kindlecomicconverter/comic2ebook.py#L170-L178

This is the behavior with the original code + the order = [2, 1] is forced: It just moves 2-5-2-5.
I don't understand why it moves to the second page and back in 2 page layout. Normally it supposed to zoom just to the first page panels, and after that the second page panels, right?

Also I have similar #PV-ML and it works fine in one page layout (landscape).

I have a Paperwhite 10 with the firmware 5.14.3.0.1. Maybe that's the problem?
How the landscape PV behavior works on your device?

**Comment by axu2** [comment]
For reference, landscape PV on my Kindle Scribe is INCREDIBLY buggy and basically unusable.

**Comment by axu2** [comment]
I just tested landscape panel view on a manga epub from humble bundle converted using Send to Kindle website. It works as expected. So it's a bug with how kcc creates the epub. @mmmeu

**Comment by VampiroMedicado** [comment]
> I just tested landscape panel view on a manga epub from humble bundle converted using Send to Kindle website. It works as expected. So it's a bug with how kcc creates the epub. @mmmeu 

Could you check which format did you receive?

I remember reading the following note:

https://www.amazon.com/gp/help/customer/display.html?nodeId=G5WYD9SAF7PGXRNA&language=en_US&currency=USD

> Note: Starting November 2023, we’ll be reducing support for sending MOBI files to your Kindle library via Send to Kindle for Email. Additionally, we’ll end support for sending MOBI files to your Kindle library through any Kindle apps on iOS or Android, or Send to Kindle apps on the desktop. This change won't affect any MOBI files already in your Kindle library. You can still read them with Kindle. MOBI is an older file format and won't support the newest Kindle features for document.

**Comment by axu2** [comment]
@VampiroMedicado I sent a humble bundle epub. I received an azw3 file ony kindle scribe.

**Comment by VampiroMedicado** [comment]
> @VampiroMedicado I sent a humble bundle epub directly using the send to Kindle website, not involving kcc. I received an azw3 file on my kindle scribe. 
> 
> Epub is an intermediate format that is fed into kindlegen. 

Yeah I understand the steps, I was mostly curious of the format.

And I guess you checked, but if you send that same EPUB as source to convert to MOBI it works fine (through KCC)?

**Comment by axu2** [comment]
@mmmeu try adding the code from the reverted 

- #499

To see if it fixes. Disable panel view enables virtual panels like humble bundle epubs.

**Comment by mmmeu** [comment]
@axu2 I tried it and there were neither. Cbz to Mobi.

**Comment by axu2** [comment]
> @axu2 I tried it and there were neither. Cbz to Mobi.

Did you activate virtual panels in the Aa menu?

**Comment by mmmeu** [comment]
@axu2 No, I'm dumb. Now it works. Thanks for your help! In the meantime I upgraded to 5.16.2.1.1. However it only does 4 virtual panels, not 6, even if vertical-rl is added and Synthetic Spread is enabled. How much are the virtual panels customizable? Fit to width etc.?
